iotdb/src/main/java/org/apache/iotdb/db/query/executor/EngineExecutorWithoutTimeGenerator.java

```diff
@@ -71,33 +71,50 @@ public QueryDataSet executeWithGlobalTimeFilter()
       QueryDataSource queryDataSource = QueryDataSourceManager.getQueryDataSource(jobId, path);
 
       // add data type
-      dataTypes.add(MManager.getInstance().getSeriesType(path.getFullPath()));
+      try {
+        dataTypes.add(MManager.getInstance().getSeriesType(path.getFullPath()));
+      } catch (PathErrorException e) {
+        throw new FileNodeManagerException(e);
+      }
 
       PriorityMergeReader priorityReader = new PriorityMergeReader();
 
       // sequence reader for one sealed tsfile
-      SequenceDataReader tsFilesReader = new SequenceDataReader(queryDataSource.getSeqDataSource(),
-          timeFilter);
-      priorityReader.addReaderWithPriority(tsFilesReader, 1);
+      SequenceDataReader tsFilesReader = null;
+      try {
+        tsFilesReader = new SequenceDataReader(queryDataSource.getSeqDataSource(),
+            timeFilter);
+        priorityReader.addReaderWithPriority(tsFilesReader, 1);
+      } catch (IOException e) {
+        throw new FileNodeManagerException(e);
+      }
 
       // unseq reader for all chunk groups in unSeqFile
-      PriorityMergeReader unSeqMergeReader = SeriesReaderFactory.getInstance()
-          .createUnSeqMergeReader(queryDataSource.getOverflowSeriesDataSource(), timeFilter);
-      priorityReader.addReaderWithPriority(unSeqMergeReader, 2);
+      PriorityMergeReader unSeqMergeReader = null;
+      try {
+        unSeqMergeReader = SeriesReaderFactory.getInstance()
+            .createUnSeqMergeReader(queryDataSource.getOverflowSeriesDataSource(), timeFilter);
+        priorityReader.addReaderWithPriority(unSeqMergeReader, 2);
```

Comment:
We can replace it with an Enum class, but it is really semantic-related and may be fixed by the primary author

Reply:
@jixuan1989

Reply:
Replacing it with an Enum class is not enough. Because to merge sort un-sequence files, there may need lots of priority number.

I have added comments for this method.
```
 /**
   * The bigger the priority value is, the higher the priority of this reader is
   */
```

Reply:
@Beyyes Thanks for your comments, it's good enough for this pr. Nevertheless, I still intend to use `Enum` to clarify the priority level. The priority number is too simple and easy to get out of control. `Enum` format may have two advantages:

1. `Enum` forces developers to think carefully about the full structure and brings clearer design.
2. It‘s also good for readers and other developers to understand how many priorities that IoTDB has.

I'd like to discuss this issue furthermore in the future.

Reply:
Or, we can define an int array, `private static final int[] priority = new int[]{1,2,3,4}`, then readers will know the highest priority is 4. 
You can leave it in the future...

Reply:
Fixed, two final fields are set in PriorityMergeReader:
`  public static final int LOW_PRIORITY = 1;`
`  public static final int HIGH_PRIORITY = 2;`